python/pyspark/sql/functions.py

```diff
@@ -1848,6 +1848,12 @@ def __del__(self):
             self._broadcast = None
 
     def __call__(self, *cols):
+        for c in cols:
+            if not isinstance(c, (Column, str)):
+                raise TypeError(
+                    "All arguments should be Columns or strings representing column names. "
```

Comment:
"All arguments" is a little vague, since this is going to be called later in code than the UDF definition. What about an error message like "Invalid UDF argument, not a String or Column: {0} of type {1}"